packages/api/amplify_api/test/ws/web_socket_connection_test.dart

```diff
@@ -36,78 +46,165 @@ void main() {
   }''';
   final subscriptionRequest = GraphQLRequest<String>(document: graphQLDocument);
 
-  setUp(() {
+  final startAck = WebSocketMessage(
+    messageType: MessageType.startAck,
+    id: subscriptionRequest.id,
+  );
+
+  final mockSubscriptionData = {
+    'onCreateBlog': {
+      'id': '<blog id>',
+      'name': 'Unit Test Blog',
+      'createdAt': '2022-09-12T18:03:36.230Z'
+    }
+  };
+
+  final mockSubscriptionEvent = {
+    'id': subscriptionRequest.id,
+    'type': 'data',
+    'payload': {
+      'data': mockSubscriptionData,
+    },
+  };
+
+  WebSocketChannel getWebSocketChannel(Uri uri) {
+    return MockWebSocketChannel();
+  }
+
+  MockWebSocketConnection getWebSocketConnection() {
```

Comment:
suggestion/question: Could this get moved to test util so you can use in dart_graphql_test.dart too?

Reply:
I moved some of the logic, but can't completely reuse this logic as each test is implemented slightly different.